### PR TITLE
SAQ-295 Remove missing variables properly and remove regex

### PR DESCRIPTION
**Pull request by jrwishart**
Entirely missing variables where removed using a regex. The regex was
used to catch the missing variables when they appear in the formula for
interaction terms. e.g. x_missing + x_missing:x_int removes both terms.
This approach removed variables that weren't missing due to the
aggressive regex. This commit removes the entirely missing variables
without using any regex.

### CODE REVIEW

File: R/regression.R [hunk 1]
```diff
@@ -1600,31 +1604,33 @@ setChartData <- function(result, output)
...
+#' @importFrom flipU OutcomeName
+removeMissingVariables <- function(data, formula, interaction, missing.variables)
 {
-    missing.variable.names <- colnames(data)[missing.variables]
+    all.variables <- names(missing.variables)
+    missing.variable.names <- all.variables[missing.variables]
+    outcome.name <- OutcomeName(formula)
     if (outcome.name %in% missing.variable.names)
-        stop("Response variable is entirely missing (all observed values of the variable are missing).")
+        stop("Outcome variable is entirely missing (all observed values of the variable are missing).")
```
Comment (jrwishart) [inline R/regression.R]: Using outcome terminology here to be consistent with the app.

File: R/regression.R [hunk 2]
```diff
@@ -1600,31 +1604,31 @@ setChartData <- function(result, output)
     return(result)
 }
 
-removeMissingVariables <- function(data, formula, formula.with.interaction,
-                                   missing.variables, outcome.name, input.formula)
+#' @importFrom stats reformulate update
+#' @importFrom flipU OutcomeName
+removeMissingVariables <- function(data, formula, interaction, missing.variables)
```
Comment (JustinCCYap) [inline R/regression.R:1609]: The 3rd argument used to be a formula and now it is a character of the name of the interaction variable. Also `outcome.name` and `input.formula` are gone. Do we need to update calls to `removeMissingVariables`?

**Review by JustinCCYap — APPROVED** [review]
LGTM, see comment

File: R/regression.R (hunk 2 above)
Comment (jrwishart) [reply R/regression.R:1609]: It has been updated in `regression.R` at https://github.com/Displayr/flipRegression/blob/SAQ-295/R/regression.R and doesnt appear elsewhere.

**Pull request merged**